src/transaction.ts

```diff
@@ -458,3 +492,215 @@ export class TransactionHash extends Hash {
+      this.guardianSignature = new Uint8Array();
+    }
+}
+
+/**
+ * An utilitary class meant to work together with the {@link TransactionNext} class.
+ */
+export class TransactionComputer {
+  constructor() {}
+
+  computeTransactionFee(transaction: ITransactionNext, networkConfig: INetworkConfig): BigNumber{
+    let moveBalanceGas = new BigNumber(
+      networkConfig.MinGasLimit + transaction.data.length * networkConfig.GasPerDataByte);
+    if (moveBalanceGas > transaction.gasLimit) {
+      throw new errors.ErrNotEnoughGas(parseInt(transaction.gasLimit.toString(), 10));
+    }
+
+    let gasPrice = new BigNumber(transaction.gasPrice);
+    let feeForMove = moveBalanceGas.multipliedBy(gasPrice);
+    if (moveBalanceGas === transaction.gasLimit) {
+      return feeForMove;
+    }
+
+    let diff = new BigNumber(transaction.gasLimit).minus(moveBalanceGas);
+    let modifiedGasPrice = gasPrice.multipliedBy(
+      new BigNumber(networkConfig.GasPriceModifier)
+    );
+    let processingFee = diff.multipliedBy(modifiedGasPrice);
+
+    return feeForMove.plus(processingFee);
+  }
+
+  computeBytesForSigning(transaction: ITransactionNext): Uint8Array{
+    let plainTransaction = this.toPlainObject(transaction);
+
+    if (plainTransaction.signature) {
+      delete plainTransaction.signature;
+    }
+
+    if (plainTransaction.guardianSignature) {
+      delete plainTransaction.guardianSignature;
+    }
+
+    if (!plainTransaction.guardian) {
+      delete plainTransaction.guardian
+    }
+
+    const serialized = JSON.stringify(plainTransaction);
+
+    return new Uint8Array(Buffer.from(serialized));
+  }
+
+  computeTransactionHash(transaction: ITransactionNext): Uint8Array{
+    let serializer = new ProtoSerializer();
+
+    const tx = Transaction.fromTransactionNext(transaction);
+    let buffer = serializer.serializeTransaction(tx);
+    let hash = createTransactionHasher(TRANSACTION_HASH_LENGTH)
+      .update(buffer)
+      .digest("hex");
+    
+    return new Uint8Array(Buffer.from(hash, "hex"));
+  }
+
+  private toPlainObject(transaction: ITransactionNext) {
+    return {
+      nonce: Number(transaction.nonce),
+      value: new BigNumber(transaction.value).toFixed(0),
+      receiver: transaction.receiver,
+      sender: transaction.sender,
+      senderUsername: transaction.senderUsername ? Buffer.from(transaction.senderUsername).toString("base64") : undefined,
+      receiverUsername: transaction.receiverUsername ? Buffer.from(transaction.receiverUsername).toString("base64") : undefined,
+      gasPrice: Number(transaction.gasPrice),
+      gasLimit: Number(transaction.gasLimit),
+      data: transaction.data.length == 0 ? undefined : Buffer.from(transaction.data).toString("base64"),
+      chainID: transaction.chainID,
+      version: transaction.version,
+      options: transaction.options == 0 ? undefined : transaction.options,
+      guardian: transaction.guardian == "" ? undefined : transaction.guardian,
```

Comment:
Check can also be: `transaction.guardian ? undefined : transaction.guardian`, so that it handles `null` values, as well. CC: @CiprianDraghici, makes sense? I think for `options`, as well. Since setters are public.

Reply:
Check is now:
```js
transaction.guardian ? transaction.guardian : undefined
```

Same for `options`.